packages/app/src/navigation/SidebarNavigation.vue

```diff
@@ -165,9 +165,17 @@ const NAV_EXPAND_MIN_SCREEN_WIDTH = 1024
 
 const loginConnectStore = useLoginConnectStore()
 
-const debugBadge = computed<Badge | undefined>(() => {
-  if (props.isLoading) {
-    return undefined
+const debugBadge = ref<Badge | undefined>()
+
+const setDebugBadge = useDebounceFn((badge) => {
+  debugBadge.value = badge
+}, 500)
+
+watchEffect(() => {
+  if (props.isLoading && loginConnectStore.project.isProjectConnected) {
+    setDebugBadge(undefined)
```

Comment:
It's probably obvious and I missed it, but why do we need to debounce this? Is it to give the requests enough time so it doesn't flicker the badge on the side?